File: src/policy/loadPolicy.ts
import { readFileSync } from 'fs';
import { parse } from 'yaml';
import { Policy, type PolicyConfig } from './schema.js';
import { logger } from '../utils/logging.js';

export function loadPolicy(path: string = 'config/model-policy.yaml'): PolicyConfig {
  try {
    const fileContent = readFileSync(path, 'utf8');
    const parsed = parse(fileContent);
    const validated = Policy.parse(parsed);
    
    logger.info('✅ Policy loaded from', path);
    return validated;
  } catch (error) {
    logger.error('Failed to load policy:', error);
    throw new Error(`Failed to load policy from ${path}: ${error}`);
  }
}

File: jarvis/policy/load_policy.py
"""Ported from src/policy/loadPolicy.ts."""
from __future__ import annotations

import yaml

from ..utils.logging import logger
from .schema import PolicyConfig

DEFAULT_POLICY_PATH = "config/model-policy.yaml"


def load_policy(path: str = DEFAULT_POLICY_PATH) -> PolicyConfig:
    try:
        with open(path, "r", encoding="utf-8") as f:
            parsed = yaml.safe_load(f)
        validated = PolicyConfig.model_validate(parsed)
        logger.info("✅ Policy loaded from", path)
        return validated
    except Exception as error:  # noqa: BLE001 -- surfaced immediately below
        logger.error("Failed to load policy:", error)
        raise RuntimeError(f"Failed to load policy from {path}: {error}") from error
